src/context.h

```diff
@@ -187,6 +185,12 @@ struct ly_ctx;
                                         directory, which is by default searched automatically (despite not
                                         recursively). */
 #define LY_CTX_PREFER_SEARCHDIRS 0x20 /**< When searching for schema, prefer searchdirs instead of user callback. */
+#define LY_CTX_SET_PRIV_PARSED 0x40 /**< For all compiled nodes, their private objects (::lysc_node.priv) are used
```

Comment:
`ly_ctx_destroy()` should also reflect the new option. I would not blindly expect that it will be called without the destructor, let's just ignore the destructor callback in case this option is in use.

Reply:
I also took it into consideration, but currently the destructor for private pointers is not used anyway.
[see here](https://github.com/CESNET/libyang/blob/619dc2388d3f76355842a21c1e56c2c098339ef2/src/tree_schema_free.c#L954)
But I should have added it there at least some if statement. So I'll add it.

Reply:
ok, after reviewing the mechanism of freeing `priv` pointers, it seems better to remove it from the context destructor, but let's make it in a separate PR #1501.